static-assets/components/cstudio-dialogs/new-content-type.js

```diff
@@ -173,13 +187,16 @@ CStudioAuthoring.Dialogs.NewContentType = CStudioAuthoring.Dialogs.NewContentTyp
       }
     });
 
+    // 'Display Label' input
     YEvent.on('contentTypeDisplayName', 'keyup', function () {
       YAHOO.Bubbling.fire('content-type.values.changed');
       value = document.getElementById('contentTypeDisplayName').value;
 
       value = value.replace(/[^a-z0-9]/gi, '');
       value = value.toLowerCase();
 
+      self.onSetDirty(Boolean(value));
```

Comment:
Should avoid setting this every keystroke. Set only when it changes from blank to not-blank and backwards and/or make sure react-side is also not setting state on every keystroke.